### PR TITLE
fixed the lineage tabs displayed

### PR DESCRIPTION
**Pull request by MihaiIliescu**
Signed-off-by: Iliescu Cristian-Mihai <cristianmihaiiliescu@gmail.com>

### CODE REVIEW

File: src/components/Lineage/Graph/index.tsx [hunk 1]
```diff
@@ -74,16 +79,25 @@ export function EgeriaLineageGraph(props: Props) {
   };
 
   useEffect(() => {
+
+    if (!Object.values(LINEAGE_TYPES).includes(lineageType)) {
```
Comment (sarbull) [inline src/components/Lineage/Graph/index.tsx:83]: i believe this check should be done after the "fetchData" brought the data with the selectedNode, this way you will have the Node Type available and check if it has a lineageType

File: src/components/Lineage/Graph/index.tsx [hunk 2]
```diff
@@ -94,63 +108,67 @@ export function EgeriaLineageGraph(props: Props) {
...
-            { !loading && (rawData.nodes.length > 0) && <HappiGraph rawData={{...rawData}}
-                            algorithm={'VISJS'}
-                            debug={false}
-                            graphDirection={'HORIZONTAL'}
-                            selectedNodeId={guid}
-                            actions={ <EgeriaLineageGraphActions rawData={{...rawData}} printUri={printUri}/> }
-                            onNodeClick={(d: any) => { setSelectedNodeData(d); setOpened(true); }} /> }
+          <Tabs.Panel value={LINEAGE_TYPES.ULTIMATE_SOURCE}>
+            {!loading && (rawData.nodes.length === 0) && <EgeriaCantDisplay />}
+            {!loading && (rawData.nodes.length > 0) && <HappiGraph rawData={{ ...rawData }}
```
Comment (sarbull) [inline src/components/Lineage/Graph/index.tsx:151]: not consistent with the whitespaces here, in some places you removed them, in some other place you added, please decide

**Review by sarbull — CHANGES_REQUESTED** [review]

File: src/components/Lineage/Graph/index.tsx (hunk 2 above)
Comment (MihaiIliescu) [reply src/components/Lineage/Graph/index.tsx:151]: It was done by the auto formatting, I've seen that you used rawData={{...rawData }} formatting while it reccomends rawData={{ ...rawData }} instead. I'll stick with yours, looks better

File: src/components/Lineage/Graph/index.tsx (hunk 1 above)
Comment (MihaiIliescu) [reply src/components/Lineage/Graph/index.tsx:83]: if this check is done after the fetch then the fetch will fail due to the fact that the link is wrong

File: src/components/Lineage/Graph/index.tsx (hunk 2 above)
Comment (MihaiIliescu) [reply src/components/Lineage/Graph/index.tsx]: fixed

File: src/components/Lineage/Graph/index.tsx (hunk 1 above)
Comment (sarbull) [reply src/components/Lineage/Graph/index.tsx:83]: how do you know the link is wrong?

Comment (MihaiIliescu) [reply src/components/Lineage/Graph/index.tsx:83]: lineageType comes as a path parameter and if for example you put vertical-lineageee instead of vertical-lineage in the URL, the call will fail

Comment (MihaiIliescu) [reply src/components/Lineage/Graph/index.tsx:83]: const uri = (lineageType: any) => `/api/lineage/entities/${guid}/${lineageType}?includeProcesses=true`;

**Review by lpalashevski — APPROVED** [review]

File: src/components/Lineage/Graph/RouteWrapper/index.tsx [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
+import { useParams, useLocation, useNavigate } from 'react-router-dom';
 import { EgeriaLineageGraph } from '..';
 
+
 export function EgeriaLineageGraphRouteWrapper() {
-  const { guid, lineageType } = useParams();
+  const { guid } = useParams();
+  const location = useLocation();
+
+  const lineageType = location.pathname.substring(location.pathname.lastIndexOf('/') + 1);
```
Comment (sarbull) [inline src/components/Lineage/Graph/RouteWrapper/index.tsx:9]: this needs a bit more attention, testing, the user might enter the url with "/" at the end, unit tests would be good to check the corner cases

**Review by sarbull — APPROVED** [review]
